python/tchannel/zipkin/tracers.py

```diff
@@ -126,27 +133,25 @@ class TChannelZipkinTracer(object):
     buffering of any sort.
     """
 
-    def __init__(self, tchannel):
+    def __init__(self, tchannel, routers):
         """
         :param tchannel:
             A tchannel instance to send the trace info to zipkin server
+        :param routers:
+            A list contains hyperbahn instances' ip addresses
         """
         self._tchannel = tchannel
-        self.client = tchannel.request(
-            "tcollector"
-        )
+        self.client = TCollectorClient(self._tchannel)
+
+        if routers:
+            hyperbahn.advertise(self._tchannel, 'tcollector', routers)
 
     def record(self, traces):
         for (trace, annotations) in traces:
             try:
-                self.client.send(
-                    InMemStream('span'),
-                    InMemStream(),
-                    InMemStream(base64_thrift_formatter(trace, annotations)),
-                    traceflag=False,  # disable tracing for span uploading
-                )
-            except:
-                pass
+                self.client.submit(thrift_formatter(trace, annotations))
```

Comment:
This is an asynchronous operation. It'll return a future. You won't get exceptions here. Add a done callback to log exceptions in case of failure. (This operation doesn't need to be a coroutine.)